manifests/charts/istio-cni/templates/daemonset.yaml

```diff
@@ -52,6 +52,10 @@ spec:
           operator: Exists
         - effect: NoExecute
           operator: Exists
+      securityContext:
+        runAsGroup: 0
+        runAsUser: 0
+        runAsNonRoot: false
```

Comment:
@bianpengyuan , 

Do we need to set the securityContext at the pod level or is it enough to set it at the container level - specifically just for the install-cni container? Can the taint-controller container run as a non-root user? In my local dev env, I can run this with the securityContext just set for the install-cni container.

kubectl get pods -n istio-system

NAME READY STATUS RESTARTS AGE
istio-cni-node-vcx94 2/2 Running 0 29s
istio-ingressgateway-d988d997f-mlftn 1/1 Running 0 29s
istiod-784cd69459-gppk9 1/1 Running 0 35s

kubectl exec -it istio-cni-node-vcx94 -n istio-system --container taint-controller -- /bin/sh
$ whoami
istio-proxy

~$ kubectl exec -it istio-cni-node-vcx94 -n istio-system --container install-cni -- /bin/sh
# whoami
root

Also, on line 130, isn't it safer to define the volume mountPath for the log dir to be `/host/var/run/istio-cni` instead of `/var/run/istio-cni` like the other mount paths (Referring to [Mounting the hostPath share in a privileged pod](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.8/html/storage/configuring-persistent-storage#persistent-storage-hostpath-pod_persistent-storage-hostpath))

Reply:
Thanks for the suggestions!. Updated to per container security context and added the host prefix.